http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/HttpOrHttp2ChannelPool.java

```diff
@@ -46,20 +45,21 @@
 
     private Promise<ChannelPool> protocolImplPromise;
     private ChannelPool protocolImpl;
+    private boolean closed;
```

Comment:
Does it need to be `volatile`?

Reply:
I don't think so, the only thread that reads/writes this is the event loop